gitHub.js

```diff
@@ -1,37 +1,100 @@
+}
+
+async function verifyUser(user) {
+    if (!user) return false;
+
+    const res = await fetch(`https://api.github.com/search/users?q=${user}`, {
+        headers: {
+            'Authorization': `Bearer ${process.env.gitHubToken}`,
+            'Accept': 'application/vnd.github+json',
+            'X-GitHub-Api-Version': '2022-11-28'
+        }
+    });
+    const data = await res.json();
+
+    if (res.ok) {
+        return (data["total_count"] >= 1) ? true : false;
+    } else if (res.status === 403) {
+        await wait(+res.headers.get("x-ratelimit-reset"));
+
+        return verifyUser(user);
+    } else {
+        console.log("GitHub API error.");
+        process.exit(1);
     }
+}
 
-    const res = await fetch(url, {
+async function getWeekAgo(user) {
+    const res = await fetch(`https://api.github.com/search/commits?q=author:${user}+committer-date:${weekAgo}..${today}`, {
         headers: {
             'Authorization': `Bearer ${process.env.gitHubToken}`,
             'Accept': 'application/vnd.github+json',
             'X-GitHub-Api-Version': '2022-11-28'
         }
     });
 
+    const data = await res.json();
+
     if (res.ok) {
-        const data = await res.json();
-        
-        if (res.headers.get("x-ratelimit-remaining") === "0") {
-            const secondsToWait = (+res.headers.get("x-ratelimit-reset") - dayjs().unix()) + 1;
-            await new Promise(resolve => setTimeout(resolve, secondsToWait * 1000));
+        return data["total_count"];
+    } else if (res.status === 403) {
+        await wait(+res.headers.get("x-ratelimit-reset"));
+
+        return getWeekAgo(user);
+    } else {
+        console.log("GitHub API error.");
+        process.exit(1);
+    }
+}
+
+async function getFourWeekAvg(user) {
+    const res = await fetch(`https://api.github.com/search/commits?q=author:${user}+committer-date:${monthAgo}..${today}`, {
+        headers: {
+            'Authorization': `Bearer ${process.env.gitHubToken}`,
+            'Accept': 'application/vnd.github+json',
+            'X-GitHub-Api-Version': '2022-11-28'
         }
+    });
+
+    const data = await res.json();
 
-        return [data["total_count"]];
+    if (res.ok) {
+        return Math.round(data["total_count"] / 4);
+    } else if (res.status === 403) {
+        await wait(+res.headers.get("x-ratelimit-reset"));
+
+        return getWeekAgo(user);
     } else {
-        const data = await res.json();
-        console.log(data.message);
+        console.log("GitHub API error.");
         process.exit(1);
     }
 }
 
+async function wait(limitReset) {
+    const secondsToWait = (limitReset - dayjs().unix()) + 1;
+    consoleOutput(`Waiting ${secondsToWait} seconds...`);
+
+    return new Promise(resolve => setTimeout(resolve, secondsToWait * 1000));
+}
+
+function consoleOutput(text) {
+    readline.cursorTo(process.stdout, 0);
+    readline.clearLine(process.stdout, 0);
+    process.stdout.write(text);
+}
```

Comment:
Would it be better to put this code in a `logger.js` file then import that into `index.js` and `gitHub.js` and use it that way?

Reply:
That would _work_, but from an architecture perspective this should be coming from the CLI module because that's where all the rest of the terminal input/output is.